cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE DerivingVia #-}
+
+module Cardano.Api.IPC.Monad
+  ( LocalStateQueryScript
+  , sendMsgQuery
+  , setupLocalStateQueryScript
+  ) where
+
+import Cardano.Api.Block
+import Cardano.Api.IPC
+import Control.Applicative
+import Control.Concurrent.STM
+import Control.Monad
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Cont
+import Data.Either
+import Data.Function
+import Data.Maybe
+import Data.Ord
+import Shelley.Spec.Ledger.Scripts ()
+import System.IO
+
+import qualified Ouroboros.Network.Protocol.LocalStateQuery.Client as Net.Query
+import qualified Ouroboros.Network.Protocol.LocalStateQuery.Type as Net.Query
+
+newtype LocalStateQueryScript block point query r m a = LocalStateQueryScript
+  { runLocalStateQueryScript
+      :: (a -> m (Net.Query.ClientStAcquired block point query m r))
+      -> m (Net.Query.ClientStAcquired block point query m r)
+  } deriving (Functor, Applicative, Monad, MonadIO) via ContT (Net.Query.ClientStAcquired block point query m r) m
+
+sendMsgQuery :: Monad m => query a -> LocalStateQueryScript block point query r m a
+sendMsgQuery q = LocalStateQueryScript $ \f -> pure $
+  Net.Query.SendMsgQuery q $
+    Net.Query.ClientStQuerying
+    { Net.Query.recvMsgResult = f
+    }
+
+setupLocalStateQueryScript ::
+     STM x
```

Comment:
What is `x`?

Reply:
The `x` is unknown.  I don't care about the type of the value.  See `void $ atomically waitDone` later in the function.  All I need is to know that the value exists, but I don't care what type it actually is.  The fact that I can read the value indicates everything is complete and I can exit.